webpack/scenes/ContentViews/Table/ContentViewTable.js

```diff
@@ -0,0 +1,140 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Table,
+  TableHeader,
+  TableBody,
+} from '@patternfly/react-table';
+import PropTypes from 'prop-types';
+
+import Loading from './Loading';
+import tableDataGenerator, { buildColumns } from './tableDataGenerator';
+import emptyRows from './emptyRows';
+import './ContentViewTable.scss';
+
+const contentViewShowFakeData = require('../data/show');
+
+const ContentViewTable = ({ contentViews }) => {
+  const [detailsMap, setDetailsMap] = useState({}); // Map of CV id to details object
+  const [table, setTable] = useState({ rows: [], columns: [] });
+  // Map of CV id to expanded cell, if id not present, row is not expanded
+  const [expandedColumnMap, setExpandedColumnMap] = useState({});
+  const cvsPresent = contentViews && contentViews.results && contentViews.results.length > 0;
+
+  useEffect(
+    () => {
+      if (!contentViews) return;
+      if (cvsPresent) {
+        const tableData = tableDataGenerator(
+          contentViews,
+          detailsMap,
+          expandedColumnMap,
+        );
+        setTable(tableData);
+      } else {
+        setTable({ columns: buildColumns(), rows: emptyRows });
+      }
+    },
+    [contentViews, detailsMap, expandedColumnMap],
```

Comment:
if you are not familiar with React hooks, the effect will only be run if these variables change

Reply:
`contentViews` will cover any updates there